### PR TITLE
Add pointer cursor to menu button.

### LINKED ISSUE #3031: Navigation: The burger menu button should have `cursor: pointer` when hovered over.
**Describe the bug**
The burger menu button should have `cursor: pointer` when hovered over. It currently does not. This button is not used on breakpoints > 599px, so it's not expected to be seen a great deal on a desktop.

**To Reproduce**
Hover over the burger menu button.

1. Go to any page with a screen width < 600px;
2. Hover your mouse over the burger menu button.

**Expected behaviour**
The burger menu button should have `cursor: pointer` when hovered over.

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Testing notes**

- [ ] This bug fix is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by Bopchy**
Resolves #3031 

**Overall change:** _Add pointer cursor to hamburger menu button._

**Code changes:**

- _Add pointer cursor to hamburger menu button._
- _Update snapshots._

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Automated jest tests added (for new features) or updated (for existing features)
- [ ] This PR requires manual testing

### CODE REVIEW

File: packages/components/psammead-navigation/CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 <!-- prettier-ignore -->
 | Version | Description |
 |---------|-------------|
+| 6.0.5 | Add `cursor: pointer` to Menu button |
```
Comment (Bopchy) [inline packages/components/psammead-navigation/CHANGELOG.md]:
```suggestion
| 6.0.5 | [PR#3177](https://github.com/bbc/psammead/pull/3177) Add `cursor: pointer` to Menu button |
```

**Review by AlistairGempf — APPROVED** [review]
Looks good.

**Review by OlgaLyubin — APPROVED** [review]
LGTM

**Review by rhenshaw56 — APPROVED** [review]

**Comment by PriyaKR** [comment]
Looks good to me on storybook.

**Pull request merged**